File: API/Models/Employee.cs
namespace Models
{
    public class Employee
    {
        public string ProfileImage { get; set; }
        public string Id { get; set; }
        public string FirstName { get; set; }
        public string LastName { get; set; }

        public string? DateOfBirth { get; set; }
        public string Email { get; set; }
        public string? MobileNumber { get; set; }
        public string JoinDate { get; set; }
        public int ?JobTitleId {  get; set; }
        public string? JobTitleName { get; set; }
        public int ?LocationId { get; set; }
        public string? LocationName { get; set; }
        public int ?DepartmentId {  get; set; }
        public string? DepartmentName { get; set; }
        public int ?RoleDeptLocId { get; set; }
        public string? ManagerId { get; set; }
        public int ProjectId { get; set; }
        public string? ProjectName { get; set; }
        public int StatusId { get; set; }
        public string? StatusName { get; set; }
    }
}

File: UI/src/app/Models/employee.ts
//import { Status } from "../status/status";

import { Status } from './status';

export interface Employee {
  dateOfBirth: string;
  profileImage: string;
  firstName: string;
  lastName: string;
  email: string;
  departmentId: number;
  departmentName: string;
  locationId: number;
  locationName: string;
  jobTitleId: number;
  jobTitleName: string;
  id: string;
  joinDate: string;
  statusId: number;
  statusName: string;
  managerId: string;
  projectId: number;
  mobileNo: string;
}

// export class Employee
// {
//     profileImage:string;
//     firstName:string;
//     lastName:string;
//     email:string;
//     departmentid:number;
//     departmentName:string;
//     locationId:number;
//     locationName:string;
//     jobTitleId:number;
//     jobTitle:string;
//     id:string;
//     joinDate:string;
//     statusId:number;
//     statusId:Status;
//     managerId:string

//     constructor(employeeData:any={}){
//         this.firstName = employeeData.firstName,
//         this.lastName = employeeData.lastName,
//         this.id = employeeData.id,
//         this.profileImage=employeeData.profileImage,
//         this.departmentId = employeeData.departmentId,
//         this.jobTitleId = employeeData.jobTitle,
//         this.email = employeeData.email,
//         this.locationId = employeeData.location,
//         this.joinDate = employeeData.joinDate,
//         this.statusId = employeeData.statusId,
//         this.managerId=employeeData.managerId;
//     }
// }
